src/Features/Core/Portable/Snippets/SnippetProviders/AbstractConstructorSnippetProvider.cs

```diff
@@ -32,12 +32,25 @@ protected override async Task<TextChange> GenerateSnippetTextChangeAsync(Documen
             var syntaxFacts = document.GetRequiredLanguageService<ISyntaxFactsService>();
             var root = await document.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
             var nodeAtPosition = root.FindNode(TextSpan.FromBounds(position, position));
-            var containingType = nodeAtPosition.FirstAncestorOrSelf<SyntaxNode>(syntaxFacts.IsTypeDeclaration);
+
+            // Skip inner class in nested class if position is out of it.
+            //     For example "class Outer{\r\n ctor@ \r\n class Inner {} }"
+            // Accept even if "static class" (CS0710).
+            var containingType = nodeAtPosition.FirstAncestorOrSelf<SyntaxNode>((node) =>
+            {
+                return syntaxFacts.IsTypeDeclaration(node)
+                    && IsConstructableTypeDeclaration(syntaxFacts, node)
+                    && node.Span.Contains(position);
```

Comment:
I think the `Contains` check would work. But we have [FindTokenOnLeftOfPosition](https://sourceroslyn.io/#Microsoft.CodeAnalysis.Workspaces/J/s/src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Extensions/SyntaxTreeExtensions.cs/SyntaxTreeExtensions.cs,9bd66cba08667499,references) you could use for this.

Reply:
@genlu  Could you please explain a little more how to use FindTokenOnLeftPosition instead of Span.Contains?
In the following pattern, I tried, but I get a CloseBraceToken of InnerClass

``` csharp
$$"""
class Outer
{
    class Inner
    {
    }

    $$
}
""";
```

Reply:
@genlu   I changed it to process the parent node if CloseBraceToken is found, instead of using Span.Contains.

I tried using semantic model, but gave up on it because it only fails if the ctor is entered at the end of the file.